another PMTUD panic

This time, it seems to be unrelated to the initial maximum packet size.

```
panic: congestion BUG: decreased max datagram size from 1352 to 1302

goroutine 134676 [running]:
github.com/lucas-clemente/quic-go/internal/congestion.(*cubicSender).SetMaxDatagramSize(0xc0053c4840, 0x516)
        src/lucas-clemente/quic-go/internal/congestion/cubic_sender.go:308 +0x126
github.com/lucas-clemente/quic-go/internal/ackhandler.(*sentPacketHandler).SetMaxDatagramSize(0xc002e3b790, 0x516)
        src/lucas-clemente/quic-go/internal/ackhandler/sent_packet_handler.go:714 +0x3e
github.com/lucas-clemente/quic-go.(*session).handleHandshakeConfirmed.func1(0x516)
        src/lucas-clemente/quic-go/session.go:829 +0x4b
github.com/lucas-clemente/quic-go.(*mtuFinder).GetPing.func2(0x25e2960, 0x3324bf8)
        src/lucas-clemente/quic-go/mtu_discoverer.go:81 +0x3a
github.com/lucas-clemente/quic-go/internal/ackhandler.(*sentPacketHandler).detectAndRemoveAckedPackets(0xc002e3b790, 0xc008361700, 0x2eb66b5904, 0x32f15a0, 0x8762ae66e, 0x372c963f73, 0xc00e9ba15308e3e2, 0xc007543b90)
        src/lucas-clemente/quic-go/internal/ackhandler/sent_packet_handler.go:398 +0x2ba
github.com/lucas-clemente/quic-go/internal/ackhandler.(*sentPacketHandler).ReceivedAck(0xc002e3b790, 0xc008361700, 0xc0071a8404, 0xc00e9ba153084f22, 0x372c95aad1, 0x32f15a0, 0x1, 0x1)
        src/lucas-clemente/quic-go/internal/ackhandler/sent_packet_handler.go:293 +0xc5
github.com/lucas-clemente/quic-go.(*session).handleAckFrame(0xc003822c00, 0xc008361700, 0x25e2604, 0xc008361700, 0x0)
        src/lucas-clemente/quic-go/session.go:1364 +0x77
github.com/lucas-clemente/quic-go.(*session).handleFrame(0xc003822c00, 0x25e2620, 0xc008361700, 0xc00843eb04, 0xc001ff6b58, 0x4, 0x4, 0x1, 0x20673a0)
        src/lucas-clemente/quic-go/session.go:1200 +0xdc
github.com/lucas-clemente/quic-go.(*session).handleUnpackedPacket(0xc003822c00, 0xc006e03860, 0xc00e9ba153084f00, 0xc00e9ba153084f22, 0x372c95aad1, 0x32f15a0, 0x25, 0x5ac, 0xc006e03860)
        src/lucas-clemente/quic-go/session.go:1182 +0x56a
github.com/lucas-clemente/quic-go.(*session).handleSinglePacket(0xc003822c00, 0xc007d53680, 0xc007d536e0, 0x0)
        src/lucas-clemente/quic-go/session.go:984 +0xfab
github.com/lucas-clemente/quic-go.(*session).handlePacketImpl(0xc003822c00, 0xc007d53680, 0x6)
        src/lucas-clemente/quic-go/session.go:893 +0x85
github.com/lucas-clemente/quic-go.(*session).run(0xc003822c00, 0x0, 0x0)
        src/lucas-clemente/quic-go/session.go:614 +0xe05
github.com/lucas-clemente/quic-go.(*client).dial.func1(0xc008440e70, 0xc006abab40)
        src/lucas-clemente/quic-go/client.go:295 +0x42
created by github.com/lucas-clemente/quic-go.(*client).dial
        src/lucas-clemente/quic-go/client.go:294 +0x548

Good news: It also happened in one of the integration tests:
```
Handshake drop tests with QUIC version TLS dev version (WIP) when not using Retry using a long certificate chain: true server speaks first 
  establishes a connection when 1/3 of the packets are lost in both direction
  /home/runner/work/quic-go/quic-go/integrationtests/self/handshake_drop_test.go:236
panic: congestion BUG: decreased max datagram size

--- a/internal/ackhandler/sent_packet_history.go
+++ b/internal/ackhandler/sent_packet_history.go
@@ -64,8 +64,9 @@ func (h *sentPacketHistory) Iterate(cb func(*Packet) (cont bool, err error)) err
 // FirstOutStanding returns the first outstanding packet.
 func (h *sentPacketHistory) FirstOutstanding() *Packet {
 	for el := h.packetList.Front(); el != nil; el = el.Next() {
-		if !el.Value.declaredLost && !el.Value.skippedPacket {
-			return &el.Value
+		p := &el.Value
+		if !p.declaredLost && !p.skippedPacket && !p.IsPathMTUProbePacket {
+			return p
 		}
 	}
 	return nil

--- a/internal/ackhandler/sent_packet_history_test.go
+++ b/internal/ackhandler/sent_packet_history_test.go
@@ -87,6 +87,14 @@ var _ = Describe("SentPacketHistory", func() {
 			Expect(front).ToNot(BeNil())
 			Expect(front.PacketNumber).To(Equal(protocol.PacketNumber(2)))
 		})
+
+		It("doesn't regard path MTU packets as outstanding", func() {
+			hist.SentPacket(&Packet{PacketNumber: 2}, true)
+			hist.SentPacket(&Packet{PacketNumber: 4, IsPathMTUProbePacket: true}, true)
+			front := hist.FirstOutstanding()
+			Expect(front).ToNot(BeNil())
+			Expect(front.PacketNumber).To(Equal(protocol.PacketNumber(2)))
+		})
 	})
 
 	It("removes packets", func() {

